PyInstaller/utils/hooks/conda.py

```diff
@@ -181,80 +325,101 @@ def collect_dynamic_libs(name,
     """
     files = []
 
-    if recurse_dependencies:
-        distributions = walk_dependency_tree(name, excludes).values()
-    else:
-        distributions = [search_distribution(name)]
-
-    for distribution in distributions:
-        for file in distribution["files"]:
-            # A file is classified as a DLL if it lives inside the dedicated
-            # ``lib_dir`` DLL folder.
-            if Path(file).parent == lib_dir:
-                files.append((str(CONDA_ROOT / file), dest))
+    for file in files(name, dependencies, excludes):
+        # A file is classified as a DLL if it lives inside the dedicated
+        # ``lib_dir`` DLL folder.
+        if file.parent == lib_dir:
+            files.append((str(file.locate()), dest))
 
     return files
 
 
+# --- Map packages to distributions and vice-versa ---
+
 PYTHONPATH_PREFIXES = []
-for path in sys.path:
-    path = Path(path)
+for _path in sys.path:
+    _path = Path(_path)
     try:
-        PYTHONPATH_PREFIXES.append(path.relative_to(sys.prefix))
+        PYTHONPATH_PREFIXES.append(_path.relative_to(sys.prefix))
     except ValueError:
         pass
 
+PYTHONPATH_PREFIXES.sort(key=lambda p: len(p.parts), reverse=True)
+
 
 def _get_package_name(file):
     """Determine the package name of a Python file in ``sys.path``.
 
     :param file: A Python filename relative to Conda root (sys.prefix).
+    :type file: PackagePath
     :return: Package name or None.
 
     This function only considers single file packages e.g. ``foo.py`` or
     top level ``foo/__init__.py``\\ s. Anything else is ignored (returning
     ``None``).
     """
-    file = Path(file)
+    # TODO: In theory, a package that doesn't have an ``__init__.py`` would be
+    #       missed. I've never seen a published package which didn't have an
+    #       ``__init__.py`` but it's not mandatory so it could happen.
```

Comment:
When I first read the comment, I was confused. The only reason a package not having an `__init__` module are PEP 420 namespace packages. In the comment, please refer to these to avoid confusion, something like this:
```
TODO: Handle PEP 420 namespace packages (which are missing `__init__` module). Anyhow
no Conda PEP 420 namespace packages are known.
```